pkg/provisioner/crdprovisioner.go

```diff
@@ -46,22 +53,143 @@ import (
 
 type CrdProvisioner struct {
 	azDiskClient     azdisk.Interface
+	kubeClient       kubeClientset.Interface
 	namespace        string
 	conditionWatcher *watcher.ConditionWatcher
+	azCachedReader   CachedReader
 }
 
 const (
 	// TODO: Figure out good interval and timeout values, and make them configurable.
 	interval = time.Duration(1) * time.Second
 )
 
-func NewCrdProvisioner(azdiskClient azdisk.Interface, objNamespace string) (*CrdProvisioner, error) {
-	informerFactory := azdiskinformers.NewSharedInformerFactory(azdiskClient, consts.DefaultInformerResync)
+type CachedReader struct {
```

Comment:
Should we move the `CacheReader` definition and methods to a separate file or to utils?

Reply:
`CachedReader` usage and its definition is going to vary based on what objects the instance needs to watch; this definition at the moment implements `client.Reader` interface in a `crdProvisioner` specific way. So it would be more suitable, I think, to leave it in the crdProvisioner file for now.